app.py

```diff
@@ -1,21 +1,28 @@
+from sklearn.manifold import TSNE
+from sklearn.cluster import KMeans
+import os
+import requests
+import pandas as pd
 from flask import Flask, jsonify
 from datasets import DownloadMode, load_dataset
 
-app = Flask(__name__)
+app = Flask(_name_)
 
+API_TOKEN = os.environ['HF_SECRET_KEY']
+API_URL = "https://api-inference.huggingface.co/pipeline/feature-extraction/"
+headers = {"Authorization": f"Bearer {API_TOKEN}"}
```

Comment:
These constants could be places in the `utils.py` also, with their corresponding functions.